php/datasource/class-fieldmanager-datasource-term.php

```diff
@@ -140,54 +162,74 @@ public function preload_alter_values( Fieldmanager_Field $field, $values ) {
 	}
 
 	/**
-	 * Presave hook to set taxonomy data
-	 * @param int[] $values
-	 * @param int[] $current_values
-	 * @return int[] $values
+	 * Sort function for `get_the_terms` result set.
+	 *
+	 * @param  WP_Term $term_a First term.
+	 * @param  WP_Term $term_b Second term.
+	 * @return bool            Whether or not it is in order.
+	 */
+	public function sort_terms( $term_a, $term_b ) {
+		if ( $term_a === $term_b ) {
+			return 0;
+		}
+		return ( $term_a->term_order < $term_b->term_order ) ? -1 : 1;
+	}
+
+	/**
+	 * Presave hook to set taxonomy data.
+	 *
+	 * @param  Fieldmanager_Field $field          The base field.
+	 * @param  array              $values         The new values.
+	 * @param  array              $current_values The current values.
+	 * @return array              $values         The sanitized values.
 	 */
 	public function presave_alter_values( Fieldmanager_Field $field, $values, $current_values ) {
-		if ( !is_array( $values ) ) {
+		if ( ! is_array( $values ) ) {
 			$values = array( $values );
 		}
 
 		// maybe we can create terms here.
-		if ( get_class( $field ) == 'Fieldmanager_Autocomplete' && !$field->exact_match && isset( $this->taxonomy ) ) {
-			foreach( $values as $i => $value ) {
-				 // could be a mix of valid term IDs and new terms.
+		if ( 'Fieldmanager_Autocomplete' == get_class( $field ) && ! $field->exact_match && isset( $this->taxonomy ) ) {
```

Comment:
strict? If you have a reason for not using strict just lemme know and I'll stop dropping this comment :)